posts/2023-XX-YY-boa-release-17.md

```diff
@@ -0,0 +1,215 @@
+
+[js_promise]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/struct.JsPromise.html
+[js_regexp]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/struct.JsRegExp.html
+[js_generator]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/struct.JsGenerator.html
+[js_date]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/struct.JsDate.html
+[js_data_view]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/struct.JsDataView.html
+[wrappers]: https://docs.rs/boa_engine/0.17.0/boa_engine/object/builtins/index.html
+[try_from_js]: https://docs.rs/boa_engine/0.17.0/boa_engine/value/trait.TryFromJs.html
+[try_js_into]: https://docs.rs/boa_engine/0.17.0/boa_engine/enum.JsValue.html#method.try_js_into
+
+### New Builtins
+
+#### `Intl`
+
+Boa now has internationalization support! Although we are still working on full compliance with the
+[`ecma402` specification][402], we have a couple of `Intl` utilities in place:
+- [`Intl.Collator`][collator]
+- [`Intl.ListFormat`][list]
+- [`Intl.Locale`][locale]
+- [`Intl.Segmenter`][segmenter]
+
+However, internationalization data can be pretty expensive at times; the default data included by Boa
+is 10.6 MB in size; which is why we allow customizing the data provider used by the engine with the
+[`ContextBuilder::icu_provider`][ctx_provider] hook. For more information on how to generate custom
+internationalization data, you can check out the [**data management tutorial**][data] from [`icu4x`],
+the internationalization library used in Boa. Shoutout to the `icu4x` team, which are the ones that
+made all of this possible!
+
+Additionally, all of this is behind the `intl` feature flag, which is enabled by default but can be
+disabled to reduce Boa's binary size.
+
+[402]: https://tc39.es/ecma402/
+[collator]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator
+[list]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/ListFormat
+[locale]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Collator
+[segmenter]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter
+[provider]: https://boajs.dev/boa/doc/boa_engine/context/enum.BoaProvider.html
+[ctx_provider]: https://boajs.dev/boa/doc/boa_engine/context/struct.ContextBuilder.html#method.icu_provider
+[data]: https://github.com/unicode-org/icu4x/blob/main/docs/tutorials/data_management.md#data-management-in-icu4x
+[`icu4x`]: https://github.com/unicode-org/icu4x
+
+#### `WeakRef`, `WeakSet` and `WeakMap`
+
+We've implemented support for weak references to garbage collected objects. This allowed us to
+implement some builtins like [`WeakRef`][wr], [`WeakSet`][ws] and [`WeakMap`][wm]. Note that the
+garbage collection algorithm is always subject to future optimizations and modifications, so it is
+best to avoid using them if possible.
```

Comment:
This seems to imply that `WeakRef`, `WeakSet` and `WeakMap` shouldn't be used by JavaScript users. Is this the case?

Reply:
Yeah, that is the suggestion from [Mozilla's docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakRef#avoid_where_possible)

> Correct use of WeakRef takes careful thought, and it's best avoided if possible.

Reply:
I would reword this, since the current explanation seems to imply that the reason is future incompatibility with some optimizations, but Mozilla seems to imply that this is due to the complexity associated with using these types.